handler/src/main/java/io/netty/handler/ssl/SniHandler.java

```diff
@@ -70,7 +72,7 @@
      * @param mapping the mapping of domain name to {@link SslContext}
      */
     public SniHandler(Mapping<? super String, ? extends SslContext> mapping) {
-        this(new AsyncMappingAdapter(mapping));
+        this(new MappingAdapter(mapping));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Avoid instantiation through private constructors from outside of the constructors class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AAccessorClassGeneration)